substrate/genesis-liquidity/pallet/src/lib.rs

```diff
@@ -0,0 +1,394 @@
+    /// Remove the provided genesis liquidity for an account.
+    #[pallet::call_index(0)]
+    #[pallet::weight((0, DispatchClass::Operational))] // TODO
+    pub fn remove_coin_liquidity(origin: OriginFor<T>, balance: Balance) -> DispatchResult {
+      let account = ensure_signed(origin)?;
+      let origin = RawOrigin::Signed(GENESIS_LIQUIDITY_ACCOUNT.into());
+
+      // check we are still in genesis period
+      if Self::genesis_ended() {
+        // check user have enough to remove
+        let existing = LiquidityTokensPerAddress::<T>::get(balance.coin, account).unwrap_or(0);
+        if balance.amount.0 > existing {
+          Err(Error::<T>::NotEnoughLiquidity)?;
+        }
+
+        // remove liquidity from pool
+        let prev_sri = Coins::<T>::balance(GENESIS_LIQUIDITY_ACCOUNT.into(), Coin::Serai);
+        let prev_coin = Coins::<T>::balance(GENESIS_LIQUIDITY_ACCOUNT.into(), balance.coin);
+        Dex::<T>::remove_liquidity(
+          origin.clone().into(),
+          balance.coin,
+          balance.amount.0,
+          1,
+          1,
+          GENESIS_LIQUIDITY_ACCOUNT.into(),
+        )?;
+        let current_sri = Coins::<T>::balance(GENESIS_LIQUIDITY_ACCOUNT.into(), Coin::Serai);
+        let current_coin = Coins::<T>::balance(GENESIS_LIQUIDITY_ACCOUNT.into(), balance.coin);
+
+        // burn the SRI if necessary
+        let mut sri = current_sri.0.saturating_sub(prev_sri.0);
+        let distance_to_full_pay =
+          GENESIS_SRI_TRICKLE_FEED - Self::blocks_since_ec_security(&balance.coin);
+        let burn_sri_amount = sri.saturating_mul(distance_to_full_pay) / GENESIS_SRI_TRICKLE_FEED;
+        Coins::<T>::burn(
+          origin.clone().into(),
+          Balance { coin: Coin::Serai, amount: Amount(burn_sri_amount) },
+        )?;
+        sri -= burn_sri_amount;
+
+        // transfer to owner
+        let coin_out = current_coin.0 - prev_coin.0;
+        Coins::<T>::transfer(
+          origin.clone().into(),
+          account,
+          Balance { coin: balance.coin, amount: Amount(coin_out) },
+        )?;
+        Coins::<T>::transfer(
+          origin.into(),
+          account,
+          Balance { coin: Coin::Serai, amount: Amount(sri) },
+        )?;
+
+        // save
+        let existing = LiquidityTokensPerAddress::<T>::get(balance.coin, account).unwrap_or(0);
+        let new = existing.checked_sub(balance.amount.0).ok_or(Error::<T>::AmountOverflowed)?;
+        LiquidityTokensPerAddress::<T>::set(balance.coin, account, Some(new));
+      } else {
+        let existing = Liquidity::<T>::get(balance.coin, account).unwrap_or(0);
+        if balance.amount.0 > existing || balance.amount.0 == 0 {
+          Err(Error::<T>::NotEnoughLiquidity)?;
+        }
+        if balance.amount.0 < existing {
+          Err(Error::<T>::CanOnlyRemoveFullAmount)?;
+        }
+
+        // TODO: do external transfer instead for making it easier for the user?
+        // or do we even want to make it easier?
```

Comment:
We do want to make it easier. If they're only removing the coin, an ext transfer probably makes sense? Yet if they're removing both, it doesn't, and we have to cover that case.

I think transferring to a SRI account is fine. If they can call this fn, they have SRI and can call the next fn. The larger issue is needing SRI to call this fn in the first place.